### PR TITLE
Module4/advanced components

### PR DESCRIPTION
**Pull request by yurakhomitsky**

### CODE REVIEW

File: src/components/Dialog/Dialog.test.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+const MockComponent = () => {
+	return <div tabIndex={0}>
+		<div tabIndex={1} data-testid="mock-component">
+			Mock Component
+		</div>
+	</div>;
+};
+
+xdescribe('Dialog component tests', () => {
```
Comment (yurakhomitsky) [inline src/components/Dialog/Dialog.test.tsx:13]: I'm getting an error: Your focus-trap must have at least one container with at least one tabbable node in it at all times portal

File: src/App.tsx [hunk 2]
```diff
@@ -1,32 +1,69 @@
...
+						<h2 className="search-field-title">Find your movie</h2>
+						<SearchForm searchTerm={searchTerm} onSearch={setSearchTerm} />
+					</div>
+				</Header>
+			)}
 
-		</>
-	);
+		<main>
+			<Dialog width={'60rem'} height={'40rem'}  title={'Add new Movie'} isOpen={isDialogOpen} onClose={() => setIsDialogOpen(false)}>
```
Comment (yurakhomitsky) [inline src/App.tsx]: What is the best practice to show a dialog? Currently, I manage the state by passing a boolean flag into Dialog. 

But I have seen a pattern where they manage like this`{isOpen && Dialog}` 

- In the first approach, we reduce the boilerplate a little bit but keep the component alive
- in the second approach we have a little bit of boilerplate but the component is destroyed

Comment (igor-budzin) [reply src/App.tsx]: Both approaches acceptable. Depends on requirements. I prefer to fully unmout component from DOM (`{isOpen && <Dialog />}`). But in this case you can't keep data entered by user and might be issue with CSS animation if you have such. So choose what suits to your needs

**Comment by igor-budzin** [comment]
Buttons don't work. Please fix it and I will proceed with code review
image

File: src/App.tsx (hunk 2 above)
Comment (yurakhomitsky) [reply src/App.tsx]: Okay, thank you

**Comment by yurakhomitsky** [comment]
@igor-budzin I have updated the code, please review again

**Comment by igor-budzin** [comment]
Still some issue. If I click "Add movie" appears filled in form but form should be empty in this case
image

**Comment by yurakhomitsky** [comment]
> Still some issue. If I click "Add movie" appears filled in form but form should be empty in this case image

Fixed

File: src/components/Dialog/Dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { ReactElement } from 'react';
+import { Portal } from 'react-portal';
+import styles from './Dialog.module.css';
+import classNames from 'classnames';
+import { CloseIcon } from './CloseIcon.tsx';
+import FocusTrap from 'focus-trap-react';
+
+
+interface DialogProps {
+	title: string;
```
Comment (igor-budzin) [inline src/components/Dialog/Dialog.tsx]: By requirements should be a string or JSX for title

**Comment by igor-budzin** [comment]
Date field has no value on edit movie
image

**Comment by igor-budzin** [comment]
There is an issue with story (two modals visible simultaneously)
image

**Review by igor-budzin — CHANGES_REQUESTED** [review]

File: src/components/Dialog/Dialog.tsx (hunk 3 above)
Comment (yurakhomitsky) [reply src/components/Dialog/Dialog.tsx]: Fixed

**Review by igor-budzin — APPROVED** [review]

**Pull request merged**